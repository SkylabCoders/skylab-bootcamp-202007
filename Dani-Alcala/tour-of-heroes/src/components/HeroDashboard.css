.dashboard {
    display: flex;
    flex-direction: row;
}

.top-heroes {
    background-color: blue;
    color: white;
    height: 80px;
    width: 100px;
    margin: 10px;
    display: flex;
    align-items: center;
    justify-content: center;
    
}