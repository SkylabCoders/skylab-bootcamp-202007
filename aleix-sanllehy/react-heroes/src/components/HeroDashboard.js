import React from 'react';
import heroList from '../hero.mock';
import { NavLink } from 'react-router-dom';

function HeroDashboard() {
	const maxHeroes = 4;
	const heroDashboard = heroList.slice(0, maxHeroes).map((hero) => (
		<NavLink
			key={hero.id}
			className="hero-dashboard__elements-anchor name btn btn-secondary btn-lg"
			activeClassName="my-active-class"
			to={'/hero/' + hero.id}
		>
			{hero.name}
		</NavLink>
	));
	return (
		<main className="hero-dashboard">
			<h3 className="hero-title">Top Heroes</h3>
			<div className="hero-dashboard__elements"> {heroDashboard} </div>
		</main>
	);
}

export default HeroDashboard;

//<span className="name btn btn-secondary">{hero.name}</span>
