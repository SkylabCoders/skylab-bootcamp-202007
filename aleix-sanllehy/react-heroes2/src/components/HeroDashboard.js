import React from 'react';

function HeroDashboard(props) {
	return <p>HeroDashboard works!</p>;
}

export default HeroDashboard;
