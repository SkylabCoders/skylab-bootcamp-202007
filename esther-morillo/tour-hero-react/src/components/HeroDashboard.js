import React from 'react';
import '../hero-service'
import heroList from '../hero-service';
import HeroDashboardItem from './HeroDashboardItem'


const heroTopList = heroList.slice(0, 4);

function HeroDashboard() {
	return (
		heroTopList.map((hero) => (
			<HeroDashboardItem 
			key={hero.id}
			id={hero.id}
			name={hero.name}
			/>
		))
	)
}

export default HeroDashboard;
