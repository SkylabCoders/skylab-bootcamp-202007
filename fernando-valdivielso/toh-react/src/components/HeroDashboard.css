h2 {
	margin-left: 270px;
	margin-top: 40px;
	margin-bottom: 0;
	color: rgb(136, 136, 136);
	font-family: Arial, Helvetica, sans-serif;
}

.dashboard {
	display: flex;
	flex-direction: row;
}

.dashboard__items {
	display: flex;
	align-items: center;
	justify-content: center;
	color: white;
	font-weight: 900;
	margin: 0 10px;
	background-color: rgba(15, 68, 91, 0.571);
	width: 120px;
	height: 120px;
}

* {
	font-family: Arial, Helvetica, sans-serif;
}

ul {
	list-style-type: none;
}
