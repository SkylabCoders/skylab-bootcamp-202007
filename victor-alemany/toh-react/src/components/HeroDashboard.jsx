import React, { useState, useEffect } from 'react';
import { NavLink } from 'react-router-dom';
import heroStore from '../stores/heroStore';
import { loadHeroes } from '../actions/heroActions';

function HeroDashboard(props) {
	const [heroes, setHeroes]= useState([])

    useEffect(()=>{
        heroStore.addChangeListener(onChange);
        if(heroes.length === 0) loadHeroes();
        return ()=> heroStore.removeChangeListener(onChange);
    },[heroes.length]);

    function onChange(){
        setHeroes(heroStore.getHeroes());
    }



	let myList = heroes.slice(0, 4);
	let listPromoted = myList.map((hero) => (
		<>
		<div key={hero.id} className="hero-item">
			<NavLink activeClassName='my-active-class' to={`/hero/${hero.id}`}>
				<span className="id-span">{hero.id}</span><span className="name-span">{hero.name}</span>
			</NavLink>
		</div>
		</>
	)
	);
	return (<>
		<div>
				<h1>Top Heroes</h1>
        </div>
		<div className="dashboard-item-container">
			{listPromoted}
		</div> 
	</>
	);
}
export default HeroDashboard;