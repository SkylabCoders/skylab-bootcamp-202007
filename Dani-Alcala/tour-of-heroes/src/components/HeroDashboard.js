import React from 'react';
import './HeroDashboard.css';
import { NavLink } from 'react-router-dom';


function HeroDashboard(props) {
	return <p>HeroDashboard works!</p>;
}

// function HeroDashboard({ heroes }) {
// 	return (
// 		<>
// 			<h2>Top Heroes</h2>
// 			<div className="dashboard">
// 				{heroes.map((item) => (
// 					<NavLink
// 						activeClassName="my-active-class"
// 						to="/hero/14"
// 						className="top-heroes"
// 						key={item}
// 					>
// 						{item.name}
// 					</NavLink>
// 				))}
// 			</div>
// 		</>
// 	);
// }

export default HeroDashboard;