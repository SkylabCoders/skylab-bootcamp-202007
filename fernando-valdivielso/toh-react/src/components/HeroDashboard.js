import React from 'react';
import heroData from '../heroData';
import './HeroDashboard.css';
import { NavLink } from 'react-router-dom';


function HeroDashboard(props) {

	return (
		<div >
			<h2>Top Heroes</h2>
			{fourHeroes()}
		</div>
	)


}

let four = heroData.slice(0, 4);

const fourHeroes = () => ( //por qu√© () y no {}???
	<ul className='dashboard'>
		{
			four.map(hero => (
				<NavLink to={`/hero/${hero.id}`} className='dashboard__items' key={hero.id}> {hero.name}</NavLink>
			))

		}
	</ul>
);


export default HeroDashboard;
