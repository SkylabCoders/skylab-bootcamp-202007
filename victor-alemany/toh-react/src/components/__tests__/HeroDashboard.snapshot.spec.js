import renderer from 'react-test-renderer'
import HeroDashboard from '../HeroDashboard'
import React from 'react'


describe('App snapshot', ()=>{
    const tree = renderer.create(<HeroDashboard />);

    it('should match', ()=>{
        expect(tree.toJSON()).toMatchSnapshot();
    })
})
