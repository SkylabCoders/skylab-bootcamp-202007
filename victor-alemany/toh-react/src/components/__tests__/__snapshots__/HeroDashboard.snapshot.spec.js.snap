// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot should match 1`] = `
Array [
  <div>
    <h1>
      Top Heroes
    </h1>
  </div>,
  <div
    className="dashboard-item-container"
  />,
]
`;
